File: ios/ExpoScreenCornersModule.swift
import ExpoModulesCore

public class ExpoScreenCornersModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ExpoScreenCorners")

    Events("onChange")

    Constant("PI") {
      Double.pi
    }

    Function("hello") {
      return "Hello world! 👋"
    }

    AsyncFunction("setValueAsync") { (value: String) in
      self.sendEvent("onChange", [
        "value": value
      ])
    }

    Class(ExpoScreenCornersModuleSharedObject.self) {
      Constructor { () -> ExpoScreenCornersModuleSharedObject in
        return ExpoScreenCornersModuleSharedObject()
      }

      Property("count") { (ref: ExpoScreenCornersModuleSharedObject) -> Int in
        return ref.count
      }
      .set { (ref: ExpoScreenCornersModuleSharedObject, count: Int) in
        ref.count = count
      }
    }
  }
}

File: android/src/main/java/expo/modules/screencorners/ExpoScreenCornersModule.kt
package expo.modules.screencorners

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoScreenCornersModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoScreenCorners")

    Events("onChange")

    Constant("PI") {
      Math.PI
    }

    Function("hello") {
      "Hello world! 👋"
    }

    AsyncFunction("setValueAsync") { value: String ->
      sendEvent("onChange", mapOf(
        "value" to value
      ))
    }

    Class(ExpoScreenCornersModuleSharedObject::class) {
      Constructor {
        val instance = ExpoScreenCornersModuleSharedObject(appContext)
        return@Constructor instance
      }

      Property("count")
        .get { ref: ExpoScreenCornersModuleSharedObject ->
          ref.count
        }
        .set { ref: ExpoScreenCornersModuleSharedObject, count: Int ->
          ref.count = count
        }
    }
  }
}
